rust/agama-server/src/software/web.rs

```diff
@@ -153,6 +230,74 @@ async fn products(State(state): State<SoftwareState<'_>>) -> Result<Json<Vec<Pro
     Ok(Json(products))
 }
 
+/// Information about registration configuration (product, patterns, etc.).
+#[derive(Clone, Serialize, Deserialize, utoipa::ToSchema)]
+#[serde(rename_all = "camelCase")]
+pub struct RegistrationInfo {
+    /// Registration key. Empty value mean key not used or not registered.
+    key: String,
```

Comment:
Why not use `Option<String>`?

Reply:
I am considering it, but in the end I use same values as in dbus API. Reason is mainly to prevent accidental sending just email, that can mean that system is not registered.

Reply:
I still prefer using `Option<T>` that the typical "empty means null" handling. In D-Bus we are far more limited. Of course, I am not blocking the merge because of this, just mentioning.